src/lib/CollisionPrevention/CollisionPreventionTest.cpp

```diff
@@ -804,3 +814,249 @@ TEST_F(CollisionPreventionTest, addObstacleSensorData_resolution_offset)
 		cp.getObstacleMap().distances[i] = UINT16_MAX;
 	}
 }
+
+TEST_F(CollisionPreventionTest, adaptSetpointDirection_distinct_minimum)
+{
+	// GIVEN: a vehicle attitude and obstacle distance message
+	TestCollisionPrevention cp;
+	cp.getObstacleMap().increment = 10.f;
+	obstacle_distance_s obstacle_msg {};
+	obstacle_msg.frame = obstacle_msg.MAV_FRAME_GLOBAL; //north aligned
+	obstacle_msg.increment = 10.f;
+	obstacle_msg.min_distance = 20;
+	obstacle_msg.max_distance = 2000;
+	obstacle_msg.angle_offset = 0.f;
+
+	matrix::Quaternion<float> vehicle_attitude(1, 0, 0, 0); //unit transform
+	float vehicle_yaw_angle_rad = matrix::Eulerf(vehicle_attitude).psi();
+
+	//obstacle at 0-30 deg world frame, distance 5 meters
+	memset(&obstacle_msg.distances[0], UINT16_MAX, sizeof(obstacle_msg.distances));
+
+	for (int i = 0; i < 7 ; i++) {
+		obstacle_msg.distances[i] = 500;
+	}
+
+	obstacle_msg.distances[2] = 1000;
+
+	//define setpoint
+	matrix::Vector2f setpoint_dir(1, 0);
+	float sp_angle_body_frame = matrix::atan2(setpoint_dir(1), setpoint_dir(0)) - vehicle_yaw_angle_rad;
```

Comment:
```suggestion
	float sp_angle_body_frame = matrix::atan2f(setpoint_dir(1), setpoint_dir(0)) - vehicle_yaw_angle_rad;
```

Reply:
Things prefixed with `matrix::` already get converted to float due to the funky macros over here https://github.com/PX4/Matrix/blob/973999a4d3c6d4d85bf0153230974fd9571f7846/matrix/stdlib_imports.hpp#L33-L61 , but let's rather be explicit